src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -797,8 +797,9 @@ def create_disk(size_in_mb, cloud_properties, vm_cid = nil)
         pipeline.each do |storage_placement|
           datastore = storage_placement
 
-          logger.info("Trying to create persistent disk on datastore: #{datastore.name}")
-          disk = @datacenter.create_disk(datastore, size_in_mb, disk_type)
+          disk_cid = cloud_properties.key?('name') ? cloud_properties['name'] : "disk-#{SecureRandom.uuid}"
```

Comment:
```suggestion
          disk_cid = cloud_properties['name'] || "disk-#{SecureRandom.uuid}"
```
probably shouldn't review my own pr, but this can probably be simplified:
```
irb(main):001:0> fixture = {"abc" => 123}
=> {"abc"=>123}
irb(main):002:0> fixture["abc"] || 456
=> 123
irb(main):003:0> fixture["cba"] || 456
=> 456
```